doc/tutorial/argument_parser/index.md

```diff
@@ -456,3 +456,87 @@ Add a seqan3::regex_validator to the first positional option that expects the `f
 \solution
 \snippet doc/tutorial/argument_parser/solution6.cpp file_validator
 \endsolution
+
+
+# The Version Update Feature
+
+SeqAn applications that are using the seqan3::argument_parser can check SeqAn servers for version updates.
+The functionality helps getting new versions out to users faster.
+It is also used to inform application developers of new versions of the SeqAn library
+which means that applications ship with less bugs.
+For privacy implications, please see: https://github.com/seqan/seqan3/wiki/Update-Notifications
+
+### Customized Messages
+
+The version information you receive depends on whether you are an application user or developer.
+We differentiate this by inquiring the `NDEBUG` (no-debug) macro.
+
+Case 1: `NDEBUG` is set.
+It's the default in our application and represents that you are a user.
+The only message you will eventually encounter is the following:
+
+```
+[APP INFO] :: There is a newer version of this application available.
+[APP INFO] :: If this app is developed by SeqAn, visit www.seqan.de for updates.
```

Comment:
We should definetely set a user provided URL here.

Reply:
Compromise: Can we add a `update_url` string member that gets printed here? Any tag-checking can be implemented later, but having that would already help greatly here. It's also something we can tell the developer to set in debug mode. And later, if we recognise the url-type, we won't tell people to register at all.